src/parse/spec.js

```diff
@@ -28,9 +28,10 @@ define(function(require, exports, module) {
         padding: parsePadding(spec.padding),
         signals: parseSignals(model, spec.signals),
         predicates: parsePredicates(model, spec.predicates),
-        marks: parseMarks(model, spec, width, height),
-        data: parseData(model, spec.data, function() { callback(viewFactory(model)); })
+        marks: parseMarks(model, spec, width, height)
       });
+
+      model.data(parseData(model, spec.data, function() { callback(viewFactory(model)); }))
```

Comment:
The general rationale of moving the parseData outside of the defs call is to eliminate the need for the setTimeout() call before invoking the callback.

But I'm not sure what I was thinking w.r.t. the model.data() part. I've removed that.